src/storage/mvcc/txn.rs

```diff
@@ -128,6 +128,29 @@ impl<'a> MvccTxn<'a> {
         self.unlock_key(key.clone());
         Ok(())
     }
+
+    pub fn gc(&mut self, key: &Key, safe_point: u64) -> Result<()> {
+        let mut after_safe_point = false;
+        let mut ts: u64 = u64::max_value();
+        while let Some((start, commit)) = try!(self.reader.seek_write(key, ts)) {
+            if !after_safe_point {
+                if commit <= safe_point {
+                    // Set `after_safe_point` after the latest write after `safe_point`.
+                    after_safe_point = true;
+                    // Latest write can be deleted if it's a `Modify::Delete`.
+                    if try!(self.reader.load_data(key, start)).is_none() {
```

Comment:
For a txn that has a `start_ts` greater than `safe_point`, it will be `NotExist`, which is correct because the key is deleted before `safe_point`.
For a txn that has a `start_ts` less than `safe_point`, it will be `TxnIsTooOld`, but this feature is not implemented yet...